### PR TITLE
components/modal: add modal, style and use as deleteModal for idea

### PR DESCRIPTION
**Pull request by fuzzylogic2000**
I am not sure about the reusabilty and the whole approach. But I thought any modal would have a text, an action and a cancel-thing? And if we get something else to cover, we can still change it.
Also unsure about the styling and if I would change the text-size or rather not?
I couldn't get the two lines of text centered on their own, though.

### CODE REVIEW

File: components/Modal.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      isVisible={props.isVisible}
+      containerStyle={styles.backdrop}
+    >
+      <View style={styles.list}>
+        <ListItem
+          Component={TouchableWithoutFeedback}
+          bottomDivider={true}
+          containerStyle={styles.modalText}
+        >
+          <ListItem.Content style={styles.modalContent}>
```
Comment (khamui) [inline components/Modal.js:18]: given the listitems have some things in common you could make a subcomponent something like `ModalItem`. Depending on if its not to much of conditions to set the right style classes.

File: containers/Ideas/Idea.js [hunk 2]
```diff
@@ -18,7 +19,12 @@ export const Idea = (props) => {
   };
   const toggleMenu = () => setMenuVisible(!menuVisible);
   const [menuVisible, setMenuVisible] = useState(false);
-  const menuItems  = [
+  const toggleDeleteModal = () => (
```
Comment (khamui) [inline containers/Ideas/Idea.js:22]: i know i started this mess :) (changed it in a pending PR), but lets try to have an order in our react function components.

E.g. I would start with variable and state definitions, followed by function definitions. Maybe something like the following?

`const {params, createdDate} = props.route.params;`
`const {menuVisible, setMenuVisible} = useState(false);`
`const {deleteModalVisible, setDeleteModalVisible} = useState(false);`
`const toggleMenu = () => ....`
`const toggleDeleteModal = () => ....`

File: components/Modal.styles.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { StyleSheet } from 'react-native';
```
Comment (khamui) [inline components/Modal.styles.js:1]: the font sizes are looking slightly bigger than from the menu component.

**Review by khamui — COMMENTED** [review]
Very nice!!! Working fine, just some little suggestions for improvements.

File: components/Modal.js (hunk 1 above)
Comment (khamui) [reply components/Modal.js:18]: so if that is to much for now. leave it as is (because its working fine!)

Comment (fuzzylogic2000) [reply components/Modal.js:18]: Hm, they share nearly no styling. But might still be more readable. Will try it!

File: containers/Ideas/Idea.js (hunk 2 above)
Comment (fuzzylogic2000) [reply containers/Ideas/Idea.js:22]: Ah, yes! I like rules I can stick to. :)

File: components/Modal.styles.js (hunk 3 above)
Comment (fuzzylogic2000) [reply components/Modal.styles.js:1]: Yeah, before they looked so small. Will remove the sizing again and see if @mcastro-lqd complains. :p

File: components/ModalItem.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (khamui) [inline components/ModalItem.js:1]: nice!

Comment (fuzzylogic2000) [reply components/ModalItem.js:1]: Cool, so can we merge?

**Comment by khamui** [comment]
Yes! looking very good.

**Pull request merged**